From 8c8aa5c551daf47383198189d9f50f37316d038e Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Tue, 11 Jun 2024 15:16:20 +0800
Subject: [PATCH 084/109] WIN2030-15279:fix(hf_gpio_process):bug fix of
 som_rst_feedback_process

Changelogs:
1.Judgge the return value of get_som_power_state() before doing the reset

Change-Id: I1b5bf3f00a4bab9801f21232a1fd7ef6b69d177b
---
 Core/Src/hf_gpio_process.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Core/Src/hf_gpio_process.c b/Core/Src/hf_gpio_process.c
index a970395..36ca9ff 100644
--- a/Core/Src/hf_gpio_process.c
+++ b/Core/Src/hf_gpio_process.c
@@ -153,9 +153,11 @@ static void mcu_reset_som_process(void)
 static void som_rst_feedback_process(void)
 {
 	printf("%s %d som reset\n", __func__, __LINE__);
-	som_reset_control(pdTRUE);
-	osDelay(2);
-	som_reset_control(pdFALSE);
+	if (get_som_power_state() == SOM_DAEMON_ON) {
+		som_reset_control(pdTRUE);
+		osDelay(2);
+		som_reset_control(pdFALSE);
+	}
 }
 
 #define USER_RST_THRESHOLD 10000
-- 
2.25.1

